### PR TITLE
The Particlening

### LINKED ISSUE #1793: [BUG] Having lava in your view results in FPS drops
## Reproduction:
On my fairly weak machine, I've noticed that going next to a lake of lava in places such as the lava planet will result in a major FPS drop, which I feel like https://github.com/shiptest-ss13/Shiptest/pull/1569 may have something to do with this?
Might be hard to reproduce since this will likely depend on how good your computer is.

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request

Fixes #1793

This adds a fuck ton of particles to the game. These are almost entirely ported from tgstation, specifically:
https://github.com/tgstation/tgstation/pull/71110 (I was unable to get the editor to work, so this is just the backend fancy stuff)
https://github.com/tgstation/tgstation/pull/59869 (particles)
https://github.com/tgstation/tgstation/pull/60515  (particles)
https://github.com/tgstation/tgstation/pull/74555 (particles)

And because atomization is not in my name, I have also done added my own spin on this, and that is making blood show up as a particle effect, instead of mysteriously showing up on the floor. I also made the mysteriously showing up on the floor bit more clearer if you are spilling a lot of blood.

Because of the above, I went further and ported https://github.com/DaedalusDock/daedalusdock/pull/79 (blood squirts only) as well for being shot, and some additions I made myself. I am aware using sleep() is bad practice, but I'm not sure how I would avoid using it here outside of rearranging how a lot of things work, so ill just say its "fine"

Also gibbing doesn't destroy your body anymore, however it STILL will be very hard to fix you reguardless, since you still fall apart like a lego character, it's just you no longer need to stuff them into a legion corpse.

And during testing since it annoyed me, bloodloss effects are more pronounced and harder to ignore, taken mostly from https://github.com/tgstation/TerraGov-Marine-Corps 's bloodloss effects but less severe in a way. 

And one last thing, bone breaking has sound effects, from Terragov above. Cronch.

## Why It's Good For The Game

While i'm not convinced over using particles over hand drawn sprite effects, I cannot deny i see the potential in them, especially here!

Screenshot_15726

Screenshot_15727

Screenshot_15728

Screenshot_15729

Screenshot_15730

## Changelog

:cl: Baystation12, Kapu1178, rye-rice
add: Particles!
add: Bleeding has better feedback
add: Bone breaking now has sound effects
add: Getting shot now throws blood squirts! Live through the somme for REAL this time!
balance: gibbing no longer destroys your chest, no more legion transfers!
fix: Lava particles should no longer destroy your FPS

/:cl:

### CODE REVIEW

**Comment by Apogee-dev** [comment]
I already gave my opinion about the gibbing changes on discord but I’ll repeat them here: I think it’s a move in the wrong direction. All the sources of gibbing in the game are rare, predictable, or player-initiated. If implanting brains into legion corpses is unbelievable and jank, I’d rather remove brain implanting entirely (it’s kind of absurd that random miners working out of a space ship’s bathroom can technically pull that off anyway). Require an MMI to save somebody who’s been debrained or something.  That, and leaving everything intact but scattered around sort of defeats the point of gibbing anyway, given it specifically represents somebody getting the business end of the chunky salsa rule, not neatly dismembered. 
Not all round removal is a bad thing, and some failure states- especially ones that are predictable or triggered by another player’s direct choice- should not be recoverable. Bay gets by just fine with permadeath- they have a vastly different medical system that makes dying much harder, but in theory anything _can_ permakill you, and in the end that facilitates rather than disrupts roleplay. I’m not saying we should switch to brainmed, but I am pointing out that round removal has a place even on roleplay servers.

**Comment by thgvr** [comment]
Particle updates are great. This has no reason to be bundled with the gibbing change though

**Comment by Leon9621** [comment]
As a note, I believe the bone breaking sounds here cause people to crash when they play, depending on the version of BYOND they have installed. Not sure why- wish I could be more helpful here, sorry. 

https://i.imgur.com/c9zeSqP.png pic of it happening to me, cloudbreak mentioned it happening a few times as well

File: code/datums/looping_sounds/weather.dm
Comment (thgvr) [inline code/datums/looping_sounds/weather.dm:80]:
```suggestion
		'sound/ambience/storm_outdoors.ogg' = 1,
```

File: code/modules/mob/living/blood.dm
Comment (thgvr) [inline code/modules/mob/living/blood.dm]: Then delete it

File: code/modules/surgery/bodyparts/bodyparts.dm
Comment (thgvr) [inline code/modules/surgery/bodyparts/bodyparts.dm:783]: Span macros please

File: code/modules/surgery/bodyparts/bodyparts.dm
Comment (thgvr) [inline code/modules/surgery/bodyparts/bodyparts.dm:784]: This is probably what is causing issues, maybe something about file access idk

**Review by thgvr — DISMISSED** [review]
Leave the non-visual gibbing changes (leaving a torso behind) for a different pr and discussion

**Comment by Thera-Pissed** [comment]
Lava particles can be grabbed by gravitational anomalies and flung around, which looks quite silly.

**Comment by thgvr** [comment]
We can merge this once you take out the gibbing change that leaves a torso behind

**Comment by rye-rice** [comment]
apologies for the delay in removing the anti-salt measure, got distracted by a fucking butterfly; lgtm, etc

**Pull request merged**
